src/scvi/dataloaders/_data_splitting.py

```diff
@@ -22,7 +22,11 @@
 
 
 def validate_data_split(
-    n_samples: int, train_size: float, validation_size: Optional[float] = None
+    n_samples: int,
+    train_size: float,
+    validation_size: Optional[float] = None,
+    use_external_indexing: Optional[bool] = False,
+    external_indexing: Optional[list] = None,
```

Comment:
I like the use of `external_indexing` to contain all data split indices instead of having separate arguments for training and validation. To make it more concise, we could remove `use_external_indexing` and assume that `external_indexing` will be used when it is not `None`

Reply:
We can also make the typing more explicit here with `Optional[list[npt.ArrayLike]]`

Reply:
Make sense